src/main/java/ch/njol/skript/expressions/ExprVehicle.java

```diff
@@ -99,37 +98,47 @@ public class ExprVehicle extends SimplePropertyExpression<Entity, Entity> {
 		OLD_DISMOUNT_EVENT_CLASS = oldDismountEventClass;
 		OLD_GETDISMOUNTED_HANDLE = oldGetDismountedHandle;
 	}
-	
+
+	private boolean plural;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Entity>) exprs[0]);
+		plural = parseResult.hasTag("s");
+		if (plural && getExpr().isDefault())
+			Skript.error("An event cannot contain multiple vehicles. Use 'vehicle' with no plurality in vehicle events.");
```

Comment:
error without returning false? shouldn't this be a warning then?

Reply:
I dont think that is the case. My understanding is that the parser continues to try other elements because one of them may match successfully even if this one doesnt. Either way, this should return false to indicate that intialization failed